samcli/lib/build/workflow_config.py

```diff
@@ -141,7 +141,7 @@ def supports_build_in_container(config):
     """
 
     def _key(c):
-        return c.language + c.dependency_manager + str(c.application_framework)
+        return str(c.language) + str(c.dependency_manager) + str(c.application_framework)
```

Comment:
Shouldn’t this be the hash function on the object? It’s much cleaner that way.

Reply:
The whole point of namedtuple is so you automatically get a sane hash function. I don't want to break that abstraction by defining a custom has that takes only a handful of properties. May be I am wrong here, so let me know if a custom hash is right way of doing things..